contracts/Oracle.sol

```diff
@@ -3,10 +3,68 @@
  */
 pragma solidity 0.6.10;
 
+import "./interfaces/AggregatorInterface.sol";
+import "./interfaces/AddressBookInterface.sol";
+import "./packages/oz/Ownable.sol";
+
 /**
- *
+ * @author Opyn Team
+ * @title Oracle Module
+ * @notice The Oracle module provide the system with on-chain prices
  */
-// solhint-disable-next-line no-empty-blocks
-contract Oracle {
+contract Oracle is Ownable {
+    /// @dev structure that represent price, and timestamp
+    struct Price {
+        uint256 price;
+        uint256 timestamp; // timestamp at which the price is pulled to this oracle
+    }
+
+    /// @dev mapping between oracle and it's locking period
+    mapping(address => uint256) internal oracleLockingPeriod;
+    /// @dev mapping of asset price to it's dispute period
+    mapping(address => uint256) internal oracleDisputePeriod;
+    /// @dev mapping between batch and it's oracle
+    mapping(bytes32 => address) internal batchOracle;
+    /// @dev mapping between batch and it price at specific timestmap
+    mapping(bytes32 => mapping(uint256 => Price)) internal batchPriceAt;
+
+    /**
+     * @notice get batch price
+     * @param _batch batch hash
+     * @param _timestamp price timestamp
+     * @return price and timestap at which price submitted to this contract
+     */
+    function getBatchPrice(bytes32 _batch, uint256 _timestamp) public view returns (uint256, uint256) {
+        Price memory batchPrice = batchPriceAt[_batch][_timestamp];
+        return (batchPrice.price, batchPrice.timestamp);
+    }
+
+    /**
+     * @notice get batch oracle
+     * @param _batch batch hash
+     * @return oracle address
+     */
+    function getBatchOracle(bytes32 _batch) public view returns (address) {
+        return batchOracle[_batch];
+    }
+
+    /**
+     * @notice get oracle locking period
+     * @dev during an oracle locking period, price can not be submitted to this contract
+     * @param _oracle oracle address
+     * @return locking period
+     */
+    function getOracleLockingPeriod(address _oracle) public view returns (uint256) {
+        return oracleLockingPeriod[_oracle];
+    }
 
+    /**
+     * @notice get oracle dispute period
```

Comment:
what is the oracle locking period? Is it after the oracle locking period? Better description needed

Reply:
I explained in the @dev, should I move it to @notice ?

Reply:
well like what is the locking period? When does it happen? How long is it for? What are the different time periods?

Reply:
cool, for how long and what are different time, there is no specific value, can be any value that the owner set.

Reply:
Period implies a duration of time, but here we only return `uint256` so it's vague how would that turn into a period. Maybe add something like 
```
@dev time after expiry that the oracle will be locked. During the locking period, price can not be submitted to this contract
```
And for the return pararm:`@return: time in second after expiry`

Reply:
cool will add a better description, BTW the return is not time in second after expiry, it is just a period of time, like 15min, 1 hour.. etc, not a specific time.